src/module.c

```diff
@@ -6504,8 +6545,10 @@ void moduleHandleBlockedClients(void) {
     pthread_mutex_lock(&moduleUnblockedClientsMutex);
     /* Here we unblock all the pending clients blocked in modules operations
      * so we can read every pending "awake byte" in the pipe. */
-    char buf[1];
-    while (read(server.module_blocked_pipe[0],buf,1) == 1);
+    if (listLength(moduleUnblockedClients) > 0) {
+        char buf[1];
+        while (read(server.module_blocked_pipe[0],buf,1) == 1);
+    }
```

Comment:
i'm curious, was it necessary to change this code? maybe i'm missing something.

Reply:
Not necessary, just an optimization. We can skip unnecessary `read()` call if `moduleUnblockedClients` does not contain a client. We write byte to pipe when we are adding client to `moduleUnblockedClients`. If list is empty, pipe is empty.